packages/calcite-components/src/utils/number.ts

```diff
@@ -217,3 +230,33 @@ export function expandExponentialNumberString(numberString: string): string {
 function stringContainsNumbers(string: string): boolean {
   return numberKeys.some((number) => string.includes(number));
 }
+
+/**
+ * Adds localized trailing decimals zero values to the number string.
+ * BigInt conversion to string removes the trailing decimal zero values (Ex: 1.000 is returned as 1). This method helps adding them back.
+ *
+ * @param {string} localizedValue - localized number string value
+ * @param {string} value - current value in the input field
+ * @param {NumberStringFormat} numberStringFormatter - numberStringFormatter instance to localize the number value
+ * @returns {string} localized number string value
+ */
+export function addLocalizedTrailingDecimalZeros(
+  localizedValue: string,
+  value: string,
+  formatter: NumberStringFormat
+): string {
+  let localizedDecimals;
+  const decimalSeparator = formatter.decimal;
+  if (localizedValue.includes(decimalSeparator)) {
+    localizedDecimals = localizedValue.split(".")[1];
+  }
+  const decimals = value.split(".")[1];
+
+  if (decimals && localizedDecimals !== decimals) {
+    localizedValue = localizedValue + decimalSeparator;
+    [...decimals].forEach((decimal) => {
+      localizedValue += formatter.localize(decimal);
+    });
+  }
+  return localizedValue;
+}
```

Comment:
This function is a bit confusing to me. Is it only for adding trailing 0s? If so, then why is it iterating over all of the initial value's decimal places and concat'ing them to the localizedValue? For example

value is `123.3210`
localizedValue is `123,321`

would this function return `123,321,3210`?

Also, can you use template literals instead of concatenating and add some unit tests for this function.

Reply:
Example of changing the concat'ing to template literals is this:

```ts
 if (decimals && localizedDecimals !== decimals) {
    localizedValue = localizedValue + decimalSeparator;
    [...decimals].forEach((decimal) => {
      localizedValue += formatter.localize(decimal);
    });
  }
  return localizedValue;
```
to something like:
```ts
return decimals && localizedDecimals !== decimals
    ? `${localizedValue}${decimalSeparator}${Array.from(decimals)
        .map((d) => formatter.localize(d))
        .join("")}`
    : localizedValue;
``````

Reply:
yes, it is supposed to add only trailing zeros since those are the ones that are trimmed off during localization and sanitization.  Great catch and i agree that iteration can be avoided and didn't factored in numbering systems like french :frowning_face: . Will refactor this one and add a test as well.

Reply:
To clarify, the template literals vs string concatenation thing is about weirdness with RTL langauges. Other than that the two snippets above are pretty much the same. I just use `map()` instead of `forEach` in that specific example.